dts/bindings/gpio/microchip,mcp23017.yaml

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2018-2020 Prevas A/S
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Microchip MCP23017 I2C Gpio Expander.
```

Comment:
Look at `dts/bindings/sensor/st,lis2dh-i2c.yaml` for how to have both I2C and SPI bindings but extract the common other properties up to an include file that's shared by them.